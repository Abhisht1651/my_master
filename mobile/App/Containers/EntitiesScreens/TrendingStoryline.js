// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { View, Text, TouchableOpacity, Image, ScrollView } from 'react-native'
import images from '../../Themes/Images'
import Storyline from '../../Components/StoryLineShortCardDefualt'
// Styles
import styles from './ScreenStyles/TrendingStorylineStyles'

type Props = {} & NavigationContainerProps

const TrendingStoryline = ({ navigation }: Props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={images.arrow} />
        </TouchableOpacity>
        <Text style={styles.textFont}>Trending Storylines</Text>
      </View>
      <View style={styles.contentBox}>
        <ScrollView>
          <View>
            <View style={styles.topgap} />
            <View style={styles.post}>
              <Storyline />
            </View>
            <View style={styles.gap} />
            <View style={styles.post}>
              <Storyline />
            </View>
            <View style={styles.gap} />
            <View style={styles.post}>
              <Storyline />
            </View>
            <View style={styles.gap} />
            <View style={styles.post}>
              <Storyline />
            </View>
            <View style={styles.gap} />
          </View>
        </ScrollView>
      </View>
    </View>
  )
}

export default TrendingStoryline
