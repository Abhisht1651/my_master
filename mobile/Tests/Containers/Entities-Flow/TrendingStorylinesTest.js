import React from 'react'
import { shallow } from 'enzyme'
import { Image } from 'react-native'
import { TouchableOpacity, Text } from 'react-native'
import TrendingStoryline from '../../../App/Containers/EntitiesScreens/TrendingStoryline'
// Test for Entities (snapshot and props)
describe('TrendingStoryline', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<TrendingStoryline route={{ params: { selected: [] } }} />) // no compile-time error
  })
  it('TrendingStoryline component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(1)
  })
})
