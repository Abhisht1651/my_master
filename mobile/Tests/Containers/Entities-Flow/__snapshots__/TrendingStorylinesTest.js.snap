// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingStoryline TrendingStoryline component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 100.21428571428572,
        "justifyContent": "space-between",
        "paddingLeft": 53.99999999999999,
        "paddingRight": 195.99999999999997,
        "shadowColor": "#E6ECED",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
      />
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 40,
        }
      }
    >
      Trending Storylines
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E6ECED",
      }
    }
  >
    <ScrollViewMock>
      <View>
        <View
          style={
            Object {
              "height": 62.42857142857142,
            }
          }
        />
        <View
          style={
            Object {
              "height": 529,
              "width": 600,
            }
          }
        >
          <storyLineShortCardDefault
            border={true}
            followDisplay={true}
            followerDisplay={true}
            following={true}
            friendsFollowing="45 friends Following"
            heading="World"
            onPressFollow={[Function]}
            reach={45}
            storyImage={
              Object {
                "default": "",
              }
            }
            storyTitle="UK exit from the European Union"
            trending={true}
            updatedTime="2m"
          />
        </View>
        <View
          style={
            Object {
              "height": 83.78571428571429,
            }
          }
        />
        <View
          style={
            Object {
              "height": 529,
              "width": 600,
            }
          }
        >
          <storyLineShortCardDefault
            border={true}
            followDisplay={true}
            followerDisplay={true}
            following={true}
            friendsFollowing="45 friends Following"
            heading="World"
            onPressFollow={[Function]}
            reach={45}
            storyImage={
              Object {
                "default": "",
              }
            }
            storyTitle="UK exit from the European Union"
            trending={true}
            updatedTime="2m"
          />
        </View>
        <View
          style={
            Object {
              "height": 83.78571428571429,
            }
          }
        />
        <View
          style={
            Object {
              "height": 529,
              "width": 600,
            }
          }
        >
          <storyLineShortCardDefault
            border={true}
            followDisplay={true}
            followerDisplay={true}
            following={true}
            friendsFollowing="45 friends Following"
            heading="World"
            onPressFollow={[Function]}
            reach={45}
            storyImage={
              Object {
                "default": "",
              }
            }
            storyTitle="UK exit from the European Union"
            trending={true}
            updatedTime="2m"
          />
        </View>
        <View
          style={
            Object {
              "height": 83.78571428571429,
            }
          }
        />
        <View
          style={
            Object {
              "height": 529,
              "width": 600,
            }
          }
        >
          <storyLineShortCardDefault
            border={true}
            followDisplay={true}
            followerDisplay={true}
            following={true}
            friendsFollowing="45 friends Following"
            heading="World"
            onPressFollow={[Function]}
            reach={45}
            storyImage={
              Object {
                "default": "",
              }
            }
            storyTitle="UK exit from the European Union"
            trending={true}
            updatedTime="2m"
          />
        </View>
        <View
          style={
            Object {
              "height": 83.78571428571429,
            }
          }
        />
      </View>
    </ScrollViewMock>
  </View>
</View>
`;
